normandy/control/static/control/js/components/NotificationBar.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { setNotification } from '../actions/ControlActions.js'
+import classNames from 'classnames'
+
+class NotificationBar extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      isVisible: false
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.notification) {
+      this.setState({
+        isVisible: true
+      });
+      this.setVisibilityTimer();
```

Comment:
This use of component local state and state changing in reaction to props like this makes me nervous, though it does seem like a nice API. How would you feel about managing the `isVisible` state in the Redux store? I suppose that would make the action creators more complicated. Would it make anything better/worse?